src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -119,10 +126,72 @@ private Task<CompoundProperties[]> usedCompoundsFrom(ModelSimulation simulation,
          return _compoundPropertiesMapper.MapToSnapshots(simulation.CompoundPropertiesList, project);
       }
 
-      private Task<LocalizedParameter[]> allParametersChangedByUserFrom(ModelSimulation simulation)
+      private Task<LocalizedParameter[]> allParametersChangedByUserFrom(ModelSimulation simulation, PKSimProject project)
       {
-         var changedParameters = simulation.Model.Root.GetAllChildren<IParameter>(x => x.ShouldExportToSnapshot());
-         return _parameterMapper.LocalizedParametersFrom(changedParameters);
+         var allPotentialParametersToExport = simulation.Model.Root.GetAllChildren<IParameter>(p => p.ShouldExportToSnapshot()).GroupBy(x => x.BuildingBlockType);
+         var allParametersToExport = new List<IParameter>();
+         foreach (var parametersByBuildingBlockType in allPotentialParametersToExport)
+         {
+            var onlyChangedParameters = canFilterUnchangedParameters(parametersByBuildingBlockType.Key);
+            allParametersToExport.AddRange(onlyChangedParameters ? parametersChangedFromBuildingBlock(parametersByBuildingBlockType, simulation, project) : parametersByBuildingBlockType);
+         }
+
+         return _parameterMapper.LocalizedParametersFrom(allParametersToExport);
+      }
+
+      private static bool canFilterUnchangedParameters(PKSimBuildingBlockType buildingBlockType)
+      {
+         return buildingBlockType.IsOneOf(PKSimBuildingBlockType.Compound, PKSimBuildingBlockType.Formulation, PKSimBuildingBlockType.Individual, PKSimBuildingBlockType.Event);
+      }
+
+      private IEnumerable<IParameter> parametersChangedFromBuildingBlock(IGrouping<PKSimBuildingBlockType, IParameter> parametersByBuildingBlockType, ModelSimulation simulation, PKSimProject project)
```

Comment:
Code written quickly tested but no unit test yet. This ensures that I do not write parameters that haven't changed from building block to snapshot

Reply:
Protocol excluded because we still need to refactor application to avoid parameter duplication. This should be done in next itereation in my opinion

Reply:
What about **Population**-BB? Why it's also excluded?

Reply:
I don't think we have parameter in population BB but I can add it for completion